src/table/internal.tsx

```diff
@@ -169,255 +235,252 @@ const InternalTable = React.forwardRef(
-                              stripedRows={stripedRows}
+                              wrapLines={false}
                               isEvenRow={isEven}
-                              isVisualRefresh={isVisualRefresh}
-                            />
-                          );
-                        })}
-                      </tr>
-                    );
-                  })
-                )}
-              </tbody>
-            </table>
-            {resizableColumns && <ResizeTracker />}
-          </div>
-          <StickyScrollbar
-            ref={scrollbarRef}
-            wrapperRef={wrapperRefObject}
-            tableRef={tableRefObject}
-            onScroll={handleScroll}
-          />
-        </InternalContainer>
+                              stripedRows={stripedRows}
+                              hasSelection={hasSelection}
+                              hasFooter={hasFooter}
+                              tdRef={tableCellRefs[0]}
+                              getStickyColumnProperties={() => getStickyColumnProperties(0)}
+                              style={
+                                !shouldDisableStickyColumns && (stickyColumns?.first ?? 0) > 0
+                                  ? { left: cellOffsets.first[0] }
+                                  : {}
+                              }
+                            >
+                              <SelectionControl
+                                onFocusDown={moveFocusDown}
+                                onFocusUp={moveFocusUp}
+                                onShiftToggle={updateShiftToggle}
+                                {...getItemSelectionProps(item)}
+                              />
+                            </TableTdElement>
+                          )}
+                          {visibleColumnDefinitions.map((column, colIndex) => {
+                            const isEditing =
+                              !!currentEditCell && currentEditCell[0] === rowIndex && currentEditCell[1] === colIndex;
+                            const isEditable = !!column.editConfig && !currentEditLoading;
+                            return (
+                              <TableBodyCell
+                                key={getColumnKey(column, colIndex)}
+                                style={
+                                  resizableColumns
+                                    ? {}
+                                    : {
+                                        width: column.width,
+                                        minWidth: column.minWidth,
+                                        maxWidth: column.maxWidth,
+                                      }
+                                }
+                                ariaLabels={ariaLabels}
+                                column={column}
+                                item={item}
+                                tdRef={tableCellRefs[colIndex + (hasSelection ? 1 : 0)]}
```

Comment:
I printed `tableCellRefs` and figured out that it only contains the cells of the very last row. That is because every row overrides the prev. I would change that to only assign refs to the very first row by asserting the row index. I see two reasons for that:

1. Avoiding unnecessary computations;
2. In case the last row is conditionally removed when new items are received I am not sure if the refs of will be updated.